### ISSUE TITLE
After manually deletion of a billing alert, no more plan oder apply will succeed, becuase of an 404 during plan

### ISSUE THREAD

**Issue by ckuetbach** [mongodb/terraform-provider-mongodbatlas#305]
### Terraform CLI and Terraform MongoDB Atlas Provider Version

```hcl
Terraform v0.12.21
Atlasplugin 0.6.4
```

### Terraform Configuration File 

```hcl
resource "mongodbatlas_alert_configuration" "billing_above_max" {
  project_id   =  <project-id>
  event_type = "PENDING_INVOICE_OVER_THRESHOLD"
  enabled    = true
  
  threshold = {
    threshold = 10
    operator = "GREATER_THAN"
    unit = "RAW"
  }
  
  notification {
    type_name     = "EMAIL"
    interval_min  = 60
    delay_min     = 0
    email_address  = "ckuetbach@gmail.com"
  }
  
}
```

### Steps to Reproduce

- Create a billing alert with terraform
- Delete the alert using the website
- Re Create the Resource with terraform

### Expected Behavior

- The Alert is recreated

### Actual Behavior

- The plan fails, because of an 404 (alert configuration not found)

### Additional Infos
- The statefile stored has to be manually changed in order to get the project up and running again.

- Also, if I change values like event_type or the amount at the website, terraform will not recreate the values from my configurations.

**Comment by themantissa** [comment]
@ckuetbach in repo steps was a refresh performed to refresh the state from the current system?  Could you provide debug logs so we can see the specifics?

**Comment by ckuetbach** [comment]
I don't have the debug logs ready.

But there was a `terraform refresh`, before I made the changes, because I made an update from 0.5.1 to 0.6.4 to enable billing alert feature. 

I'm not sure, how to create a debug log, do you have any documentation ready?

For me it looks, as if the provider won't check the current state correctly against the atlas API. this would explain the 404, while reading the state and also the fact it woldn't update some of the values, if changed (like amount).

I'll try to make a repro in debug mode.

**Comment by nikhil-mongo** [comment]
[debug.log](https://github.com/mongodb/terraform-provider-mongodbatlas/files/5200680/debug.log)

Here is the debug file for this issue. The Atlas API responses with 404 all the time if any other service is deleted from UI too. But, this is not expected behaviour for billing as @ckuetbach mentioned. If this happens with Atlas cluster, as an exmaple, we create a new cluster rather than showing 404 error. We definitely need to look into this.

Thanks.

**Comment by ckuetbach** [comment]
@nikhil-mongo thanks a lot. Some changes in made with the UI aren't reflected by terraform. Lets say, you change the alert in the UI from monthly to daily (with amounts), but terraform says, all is up to date.

**Comment by nikhil-mongo** [comment]
@ckuetbach As per the `eventTypeName` in the `Condition` section of Atlas [API](https://docs.atlas.mongodb.com/reference/api/alert-configurations-create-config/#request-body-parameters), the billing `event_type` is different for all the 3 options. But as we have seen that the alerts are being managed using the `alert_configuration_id` and we have seen issues managing it. So, we believe that taking care of the alert section will help us configure this as well.

Attaching the debug.log file as the `terraform refresh` does finds the change in `eventTypeName` but does not shows it as a change.

[debug.log](https://github.com/mongodb/terraform-provider-mongodbatlas/files/5203188/debug.log)

**Comment by themantissa** [comment]
@nikhil-mongo can you create an internal bug ticket for this?  PRODTRIAGE and I'll move to the appropriate queue.

**Comment by themantissa** [comment]
Internal ID: INTMDB-112

**Comment by themantissa** [comment]
Fix was included in 0.7.0, closing issue.
